game/scripts/vscripts/Sections/Chapter2/sectionCourier.ts

```diff
@@ -0,0 +1,310 @@
+        chapter2Blockades.direTopDividerCliff,
+        chapter2Blockades.direTopJungleBlocker,
+        chapter2Blockades.aboveRoshanBlocker,
+        chapter2Blockades.belowRoshanBlocker
+    ],
+    topDireT1TowerStanding: false
+}
+
+let canPlayerIssueOrders = true;
+let playerOrderMustBuyDemonEdge = false
+let playerOrderMustBuyRecipeAndCrysalis = false
+let playerOrderMustDeliverItemsFromCourier = false
+let hasPlayerRequestedToDeliverFromCourier = false
+
+const demonEdgeName = "item_demon_edge"
+const recipeName = "item_recipe_greater_crit"
+const crystalisName = "item_lesser_crit"
+const daedalusName = "item_greater_crit"
+const broadSword = "item_broadsword"
+const bladeOfAttack = "item_blades_of_attack"
+const crystalisRecipe = "item_recipe_lesser_crit"
+
+const allowedItems: Set<String> = new Set()
+
+
+const onStart = (complete: () => void) => {
+    print("Starting", sectionName);
+    CustomGameEventManager.Send_ServerToAllClients("section_started", { section: sectionName });
+
+    const radiantSecretShopLocation = Vector(-5082, 2011, 128)
+    const direSecretShopLocation = Vector(4804, 1304, 128)
+    const inFrontOfRadiantSecretShopLocation = Vector(-4840, 1822, 128)
+    const finalMovementPositionLocation = Vector(-3538, 3861, 128)
+
+    allowedItems.add(recipeName)
+    allowedItems.add(crystalisName)
+    allowedItems.add(broadSword)
+    allowedItems.add(bladeOfAttack)
+    allowedItems.add(crystalisRecipe)
+
+    const shopBtnUIPath = "HUDElements/lower_hud/shop_launcher_block/ShopCourierControls/ShopButton"
+    const demonEdgeGuideUIPath = getPathToItemInGuideByID(51)
+    const crystalisGuideUIPath = getPathToItemInGuideByID(149)
+    const daedalusGuideUIPath = getPathToItemInGuideByID(140)
+
+    canPlayerIssueOrders = true;
+    playerOrderMustBuyDemonEdge = false
+    playerOrderMustBuyRecipeAndCrysalis = false
+    playerOrderMustDeliverItemsFromCourier = false
+    hasPlayerRequestedToDeliverFromCourier = false
+
+    const playerHero = getPlayerHero();
+    if (!playerHero) error("Could not find the player's hero.");
+
+    const playerCourier = getOrError(getPlayerCourier())
+
+    const goalTracker = new GoalTracker()
+    const goalMoveToSecretShop = goalTracker.addBoolean("Move to Radiant's secret shop.")
+    const goalOpenShop = goalTracker.addBoolean("Open the shop")
+    const goalBuyDemonEdge = goalTracker.addBoolean("Buy a Demon Edge with the gold provided.")
+    const goalBuyCrystalisAndRecipe = goalTracker.addNumeric("Buy Crystalis and the Daedalus Recipe.", 2)
+    const goalRequestItemsToBeDeliveredFromCourier = goalTracker.addBoolean("Request courier to deliver the items to you.")
+    const goalWaitToCourierToDeliverItems = goalTracker.addBoolean("Wait for the courier to deliver the items to you.")
+    const goalMoveToFinalPosition = goalTracker.addBoolean("Move into the Dire jungle.")
+
+    graph = tg.withGoals(context => goalTracker.getGoals(),
+        tg.seq([
```

Comment:
(Not important:) can put this line on the previous line, then you get 1 less indent below

Reply:
Done! Indentation is love.